CSharp/ArticleProjects/TaskDependencies/Program.cs

```diff
@@ -58,105 +56,188 @@ public static void Main(string[] args)
+
+                    // Create the collection of tasks that will be added to the job.
+                    List<CloudTask> tasks = new List<CloudTask>
                     {
-                        DependsOn = TaskDependencies.OnIds("Rain", "Sun")
-                    },
+                        // 'Rain' and 'Sun' don't depend on any other tasks
+                        new CloudTask("Rain", "cmd.exe /c echo Rain"),
+                        new CloudTask("Sun", "cmd.exe /c echo Sun"),
  
-                    // Tasks 1, 2, and 3 don't depend on any other tasks. Because
-                    // we will be using them for a task range dependency, we must
-                    // specify string representations of integers as their ids.
-                    new CloudTask("1", "cmd.exe /c echo 1"),
-                    new CloudTask("2", "cmd.exe /c echo 2"),
-                    new CloudTask("3", "cmd.exe /c echo 3"),
+                        // Task 'Flowers' depends on completion of both 'Rain' and 'Sun'
+                        // before it is run.
+                        new CloudTask("Flowers", "cmd.exe /c echo Flowers")
+                        {
+                            DependsOn = TaskDependencies.OnIds("Rain", "Sun")
+                        },
  
-                    // Task 4 depends on a range of tasks, 1 through 3
-                    new CloudTask("4", "cmd.exe /c echo 4")
+                        // Tasks 1, 2, and 3 don't depend on any other tasks. Because
+                        // we will be using them for a task range dependency, we must
+                        // specify string representations of integers as their ids.
+                        new CloudTask("1", "cmd.exe /c echo 1"),
+                        new CloudTask("2", "cmd.exe /c echo 2"),
+                        new CloudTask("3", "cmd.exe /c echo 3"),
+ 
+                        // Task 4 depends on a range of tasks, 1 through 3
+                        new CloudTask("4", "cmd.exe /c echo 4")
+                        {
+                            // To use a range of tasks, their ids must be integer values.
+                            // Note that we pass integers as parameters to TaskIdRange,
+                            // but their ids (above) are string representations of the ids.
+                            DependsOn = TaskDependencies.OnIdRange(1, 3)
+                        },
+
+                        // Task 5 depends on a range of tasks, 1 through 3, and 'Flowers'
+                        new CloudTask("5", "cmd.exe /c echo 5")
+                        {
+                            DependsOn = new TaskDependencies(
+                                new[] { "Flowers" },
+                                new[] { new TaskIdRange(1, 3) })
+                        },
+
+                        // Task A is a parent task that exits with a non-zero exit code.
+                        new CloudTask("A", "cmd.exe /c echo A exit 1"),
```

Comment:
Okay, thanks. I need to look at more examples of command-line syntax as it's not something I've used in this way before.